src/main.ts

```diff
@@ -1112,6 +1114,16 @@ function generateFromJson(ctx: Context, fullName: string, messageDesc: Descripto
     messageDesc.field.filter(isWithinOneOf).filter((field) => field.oneofIndex === oneofIndex)
   );
 
+  const canonicalFromJson: { [key: string]: { [field: string]: (from: string) => Code } } = {
+    ['google.protobuf.FieldMask']: {
+      paths: (from: string) => code`typeof(${from}) === 'string'
```

Comment:
I really like the map approach; that's nudging us towards a more configuration-driven vs. the current version `if`-driven approach.

Just wondering, I wonder we're lucky here that the JSON presentation of `FieldMask` is a single string value that happens to deserialize into the `paths` field.

I.e. maybe the key of this map should be "just `FieldMask`-the type" and we should look in the map, and let it just return the whole `fromJson` body?

Basically more how like `toJSON` ended up being handled (although again I definitely like the map-driven approach)...

Wdyt?

Reply:
Indeed, I did consider doing just that. I went this way to make sure any follow up logic will also applied to fieldmasks. Admittedly, the only logic right now is the fact that we return a single JSON object (instead of constructing one field by field).

It's true that this approach only works exactly in this case, and might not work for other types (like Timestamp?).
If you feel returning the object immediately is more future proof or straightforward, I'm fine to change it.

Reply:
Ah cool, I'm good with the current approach!